### ISSUE TITLE
Something wrong with google drive

### ISSUE THREAD

**Issue by jamjimmy** [CUT3R/CUT3R#18]
Your excellent work has received a lot of attention, so you may want to consider migrating from Google Drive to another platform to avoid exceeding your download quota. Thanks！

![Image](https://github.com/user-attachments/assets/4c027fa2-a12a-4339-ad20-690aa6cf063d)

**Comment by tualatinlz** [comment]
Yes, I have the same question.

**Comment by Forrest-110** [comment]
Hi @jamjimmy @tualatinlz ! We have transferred smaller files to [huggingface](https://huggingface.co/datasets/zhangify/CUT3R_release/tree/main). Larger datasets including DL3DV, Mapfree, and Virtual KITTI2 will take longer time to transfer.

**Comment by jamjimmy** [comment]
Thank you!!!

**Comment by Gaaaavin** [comment]
@Forrest-110 I have downloaded the DL3DV depth and mask from google drive, which only contains 1K-7K. Is this the same for the huggingface version? Do you have any plans to share the 8K-11K part?

**Comment by Forrest-110** [comment]
Hi @Gaaaavin ! This is the same as huggingface version. We only processed 1K-7K subsets.

**Comment by Gaaaavin** [comment]
Thank you for the information!

**Comment by smileyenot983** [comment]
> Hi [@Gaaaavin](https://github.com/Gaaaavin) ! This is the same as huggingface version. We only processed 1K-7K subsets.

Hello! I have got an error when using the DL3DV data:
It says me that "dl3dv_loader/processed_dl3dv_ours/7K/9b75f3bcdfb2fa943d151e8db52aa597b2ebb47caa80c056c268a282c7bc4d94/dense/depth/frame_00243.npy" does not exist.

Is it ok? I see that rgb image 00243 exists, but depth does not

**Comment by Forrest-110** [comment]
> > Hi [@Gaaaavin](https://github.com/Gaaaavin) ! This is the same as huggingface version. We only processed 1K-7K subsets.
> 
> Hello! I have got an error when using the DL3DV data: It says me that "dl3dv_loader/processed_dl3dv_ours/7K/9b75f3bcdfb2fa943d151e8db52aa597b2ebb47caa80c056c268a282c7bc4d94/dense/depth/frame_00243.npy" does not exist.
> 
> Is it ok? I see that rgb image 00243 exists, but depth does not

Hi @smileyenot983! Could you please share the link from which you downloaded the data? I’d like to check it, as the data appears to be available on my end.

**Comment by smileyenot983** [comment]
> > > Hi [@Gaaaavin](https://github.com/Gaaaavin) ! This is the same as huggingface version. We only processed 1K-7K subsets.
> > 
> > 
> > Hello! I have got an error when using the DL3DV data: It says me that "dl3dv_loader/processed_dl3dv_ours/7K/9b75f3bcdfb2fa943d151e8db52aa597b2ebb47caa80c056c268a282c7bc4d94/dense/depth/frame_00243.npy" does not exist.
> > Is it ok? I see that rgb image 00243 exists, but depth does not
> 
> Hi [@smileyenot983](https://github.com/smileyenot983)! Could you please share the link from which you downloaded the data? I’d like to check it, as the data appears to be available on my end.

I have downloaded it from the links in preprocessing.md. I found out that somehow only the scene "9b75f3bcdfb2fa943d151e8db52aa597b2ebb47caa80c056c268a282c7bc4d94" got corrupted either during download, either during merge, but don't know where exactly.

**Comment by Forrest-110** [comment]
Hi @smileyenot983, this does look quite strange. I’ve uploaded that single sequence to [Hugging Face](https://huggingface.co/datasets/zhangify/CUT3R_release/resolve/main/9b75f3bcdfb2fa943d151e8db52aa597b2ebb47caa80c056c268a282c7bc4d94.tar.gz) for you to download directly.

Please let me know if you encounter issues with any other sequences.

**Comment by smileyenot983** [comment]
> Hi [@smileyenot983](https://github.com/smileyenot983), this does look quite strange. I’ve uploaded that single sequence to [Hugging Face](https://huggingface.co/datasets/zhangify/CUT3R_release/resolve/main/9b75f3bcdfb2fa943d151e8db52aa597b2ebb47caa80c056c268a282c7bc4d94.tar.gz) for you to download directly.
> 
> Please let me know if you encounter issues with any other sequences.

Thank you, I will write back ASAP. So far, I haven't seen any other issues. I really think that there was some issue on my side during one of preprocessing steps.

@Forrest-110, also one small question: How did you choose the amount of samples per dataset? Why there are so many samples from some of the datasets, but from others much less?

**Comment by Forrest-110** [comment]
Hi @smileyenot983! We keep the total number of samples fixed and adjust the dataset ratios heuristically, based on the ground-truth quality and diversity of each dataset.

**Comment by smileyenot983** [comment]
> Hi [@smileyenot983](https://github.com/smileyenot983)! We keep the total number of samples fixed and adjust the dataset ratios heuristically, based on the ground-truth quality and diversity of each dataset.

I see. @Forrest-110, I also wanted to know if you have seen any performance degradation on some of the datasets. I have prepared a subset of the data that you have used and initialized from your checkpoint(the one with image size 224) to see how the model will train. One observation is that for some reason on most of the datasets the quality increases, but on a couple(dl3dv, co3d) the quality slightly decreases. I was wondering if this is OK.

**Comment by Forrest-110** [comment]
> > Hi [@smileyenot983](https://github.com/smileyenot983)! We keep the total number of samples fixed and adjust the dataset ratios heuristically, based on the ground-truth quality and diversity of each dataset.
> 
> I see. [@Forrest-110](https://github.com/Forrest-110), I also wanted to know if you have seen any performance degradation on some of the datasets. I have prepared a subset of the data that you have used and initialized from your checkpoint(the one with image size 224) to see how the model will train. One observation is that for some reason on most of the datasets the quality increases, but on a couple(dl3dv, co3d) the quality slightly decreases. I was wondering if this is OK.

That's normal behavior. It depends on what subset you're using.

**Comment by smileyenot983** [comment]
> > > Hi [@smileyenot983](https://github.com/smileyenot983)! We keep the total number of samples fixed and adjust the dataset ratios heuristically, based on the ground-truth quality and diversity of each dataset.
> > 
> > 
> > I see. [@Forrest-110](https://github.com/Forrest-110), I also wanted to know if you have seen any performance degradation on some of the datasets. I have prepared a subset of the data that you have used and initialized from your checkpoint(the one with image size 224) to see how the model will train. One observation is that for some reason on most of the datasets the quality increases, but on a couple(dl3dv, co3d) the quality slightly decreases. I was wondering if this is OK.
> 
> That's normal behavior. It depends on what subset you're using.

Thanks for the answer. I am wondering why this happens, and if it is possible to achieve consistent error decrease on all datasets. Do you think that this happens primarily because of the balance of the datasets? i.e. DL3DV and Co3d might be too different from other datasets. Or is it because the depthmaps and poses on these datasets are noisier

**Comment by Forrest-110** [comment]
> > > > Hi [@smileyenot983](https://github.com/smileyenot983)! We keep the total number of samples fixed and adjust the dataset ratios heuristically, based on the ground-truth quality and diversity of each dataset.
> > > 
> > > 
> > > I see. [@Forrest-110](https://github.com/Forrest-110), I also wanted to know if you have seen any performance degradation on some of the datasets. I have prepared a subset of the data that you have used and initialized from your checkpoint(the one with image size 224) to see how the model will train. One observation is that for some reason on most of the datasets the quality increases, but on a couple(dl3dv, co3d) the quality slightly decreases. I was wondering if this is OK.
> > 
> > 
> > That's normal behavior. It depends on what subset you're using.
> 
> Thanks for the answer. I am wondering why this happens, and if it is possible to achieve consistent error decrease on all datasets. Do you think that this happens primarily because of the balance of the datasets? i.e. DL3DV and Co3d might be too different from other datasets. Or is it because the depthmaps and poses on these datasets are noisier

From my understanding, you're fine-tuning a pretrained checkpoint that was originally trained with a certain dataset composition, and now you're using a different dataset ratio. So I would say the primary reason is the change in dataset balance.

Additionally, DL3DV's ground truth is quite noisy, which could explain the performance drop there.

**Comment by Mollylulu** [comment]
Hi ~ Thanks for your great work! and for sharing the processed annotations. I noticed there is an RGB dir named `processed_dl3dv_ours_rgb_cam_1` in your [huggingface repo](https://huggingface.co/datasets/zhangify/CUT3R_release/tree/main/processed_dl3dv_ours_rgb_cam_1). 

I have two questions:
  1. Could you clarify the difference between these RGB images and the original ones from the official DL3DV dataset? 
  2. If I use the processed depth from your end,  is it required that I also use the corresponding RGB images from the same directory for a correct pairing?

I appreciate your help with this. Thanks!

**Comment by Forrest-110** [comment]
Hi @Mollylulu! We applied undistortion to the original images from the official DL3DV dataset. The effect turned out to be quite minor, so using the original images might also be fine — though it may be worth double-checking the alignment of the 3D points.

**Comment by Mollylulu** [comment]
Thanks for your help :) @Forrest-110
